### ISSUE TITLE
Desktop - Chat - Timezone information not updated after language change

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#4352]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

Issue was found when executing the PR #4309 

## Action Performed:
1. Login to the app using expensifail account
2. Change your language to Spanish
3. Open any chat
4. Timezone information in the chat will be shown

## Expected Result:
Timezone information in the chat gets updated right away

## Actual Result:
Timezone information in the chat gets updated only after refresh
## Workaround:
Unknown

## Platform:

Where is this issue occurring?

- Web
- Android
- Desktop App
- Mobile Web

**Version Number:**
1.0.82-0
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/127750869-d81cb2ea-b070-4feb-bd74-de807bfbd531.mp4

**Expensify/Expensify Issue URL:**

[View all open jobs on Upwork](https://www.upwork.com/ab/jobs/search/?q=Expensify%20React%20Native&sort=recency&user_location_match=2)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @nkuoch (`Engineering`), see https://stackoverflow.com/c/expensify/questions/4319 for more details.

**Comment by parasharrajat** [comment]
Should be handled by https://github.com/Expensify/App/issues/4218.

**Comment by nkuoch** [comment]
@parasharrajat The [PR](#4218) you linked to is already in staging, and is the PR that was stated as responsible in this issue description.
However, I've tried reverting it locally, and it doesn't seem to work either, so this PR is probably not responsible.
The issue on staging (and that can be reproduced locally) is actually worse than [the original issue](https://github.com/Expensify/App/issues/4218) of just not being able to refresh the timezone. Now, changing the language doesn't refresh anything on the page.
![image](https://user-images.githubusercontent.com/2463975/127829428-f0df35d9-9f2c-47dc-9c0d-861f5b346dc2.png)

**Comment by parasharrajat** [comment]
@nkuoch That PR has added few lines which is causing this issue.

Original that PR was created to show the timezone info when we change the language to other than English but after that PR is merged Now time is not changing in real-time. So literally, it should be part of that PR. 

But if you don't want it be fixed as part of that PR, I have a fix.

We need to enable prop checking in shouldComponentUpdate

**Comment by nkuoch** [comment]
If you have a fix, yes please make a new PR, I can review it, thanks!

**Comment by nkuoch** [comment]
Doing a Git bisect, [this](https://github.com/Expensify/App/pull/4253/files) seems to be the PR that introduced the bug of texts not being refreshed to the right language.

**Comment by parasharrajat** [comment]
Ok. Great find. That is going to be a regression from #4253. even if we revert the PR. I see this issue will persist bcz of this
```
    shouldComponentUpdate(nextProps, nextState) {
        return nextState.localTime !== this.state.localTime;
    }
```
It should block the update if Props changes which will block the text update on language change. Instead of this, We can simply use PureComponent or
```
    shouldComponentUpdate(nextProps, nextState) {
        return !_.isEqual(nextProps, this.props) || nextState.localTime !== this.state.localTime;
    }
```
I am submitting a PR to correct this and we can raise another issue above the bad behaviour of #4253.

**Comment by parasharrajat** [comment]
@nkuoch  https://github.com/Expensify/App/issues/4359 For reference.

**Comment by nkuoch** [comment]
@kidroca Please comment on this issue so I can assign it to you. When your PR is ready, we'll need to add the **CP Staging label** to your PR before merging it, so it gets automatically cherry picked to staging, and so the blocker label can be removed from here.

**Comment by kidroca** [comment]
@nkuoch 👍
I'm about to submit a PR, will reference this ticket as well

**Comment by roryabraham** [comment]
@kavimuru @isagoico Can we retest this in `1.0.82-3` or higher please?

**Comment by isagoico** [comment]
Retest was a pass!
